### ISSUE TITLE
[FastSpeech2] `LengthRegulator.forward()`'s max_length cannot be converted to `int` if it is `None` aka the default

### ISSUE THREAD

**Issue by SamuelLarkin** [EveryVoiceTTS/EveryVoice#290]
This hasn't caused a problem yet but the default value of `max_length` is `None` and `None` cannot be converted to `int`.

```python
In [1]: int(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 int(None)

TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

```python
class LengthRegulator(nn.Module):
    def forward(self, x, durations, max_length=None):
        repeated_list = [
            torch.repeat_interleave(x[i], durations[i], dim=0)
            for i in range(x.shape[0])
        ]
        lengths = torch.tensor([t.shape[0] for t in repeated_list]).long()
        # FIXME: int(max_length) when max_length is None is invalid
        max_length = min(lengths.max(), int(max_length))
        mask = (
            torch.arange(max_length).expand(len(lengths), max_length)
            < lengths.unsqueeze(1)
        ).to(x.device)
        out = pad_sequence(repeated_list, batch_first=True, padding_value=0)
        if max_length is not None:
            out = out[:, :max_length]
        return out, mask
```

**Comment by roedoejet** [comment]
I don't think this is a real issue that will cause any real problems since max_length is always provided, but in some weird situation where someone is using this code and does not provide max_length, the error will be cryptic.

TLDR; I think we should close this or just produce a more transparent error if `max_length == None`
